### DISCUSSION TITLE
Handle "Credits" when Listings is turned off as an extension

### DISCUSSION THREAD

**Discussion by amywtlin** [forem/forem#16749] [Feature Requests]
**Context**

We are [moving "Listings" behind a feature flag](https://github.com/forem/forem/pull/16437) as it is not a feature that all Forem creators would use for their Forems. While Credits are only used for listings as of now so we should also hide it, @jeremyf brought up a good point that we cannot guarantee we won't use credits for more things in the future. If we hide credits automatically when Listings is turned off, this implicitness could also cause confusion. Hence I think we should handle this explicitly. This brings about a bit of communication challenge...Here's my proposal of how to solve it, and I'd like feedback and suggestions how else you would solve it.

**Goal**
1. Make admins understand what's going on
2. Make it easy for admins to toggle extensions on/off with a complete user experience

**Proposal**
1. When Listings is turned off, we prompt something on the interface, encourage admins to also turn off Credits and the reason why; if Credits is turned off, it is the same as moving Credits behind a feature flag.
2. Admins can then decide whether they want to also turn off Credits

**Timeline**
I'd like to finalize the approach of this by end of next week, and will close the feedback period by next Wed (3/9). Thanks!

**Comment by JennieOcken** [comment]
I like that as an approach, it gives us the maximum flexibility for future improvements while making it clear to admins the consequences.  My goal is to have these off by default which will help accidental usage.

**Comment by amywtlin** [comment]
@Ridhwana As I would like to move this forward by closing the discussion this Wed (3/9), I would love to know if you have any feedback on the approach!

**Reply by Ridhwana** [reply]
@amywtlin just caught up on this now. I think this sounds good.
